addons/crm_livechat/models/chatbot_script_step.py

```diff
@@ -103,10 +103,10 @@ def _process_step_create_lead_and_forward(self, discuss_channel):
         )[discuss_channel.livechat_channel_id]
         message = self._process_step_forward_operator(discuss_channel, users=users)
         operator_partner = discuss_channel.livechat_operator_id
-        if operator_partner != self.chatbot_script_id.operator_partner_id:
-            user = next(user for user in users if user.partner_id == operator_partner)
+        if users and teams and operator_partner != self.chatbot_script_id.operator_partner_id:
+            user = next((user for user in users if user.partner_id == operator_partner), self.env.user)
```

Comment:
`self.env.user` (which is the visitor) makes no sense as lead assignee, the default should be to assign no-one